components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/tokenprocessor/DefaultRefreshTokenGrantProcessor.java

```diff
@@ -0,0 +1,181 @@
+            throws IdentityOAuth2Exception {
+
+        Timestamp timestamp = new Timestamp(new Date().getTime());
+        String tokenId = UUID.randomUUID().toString();
+
+        AccessTokenDO accessTokenDO = new AccessTokenDO();
+        accessTokenDO.setConsumerKey(tokenReq.getClientId());
+        accessTokenDO.setAuthzUser(tokReqMsgCtx.getAuthorizedUser());
+        accessTokenDO.setScope(tokReqMsgCtx.getScope());
+        accessTokenDO.setTokenType(tokenType);
+        accessTokenDO.setTokenState(OAuthConstants.TokenStates.TOKEN_STATE_ACTIVE);
+        accessTokenDO.setTokenId(tokenId);
+        accessTokenDO.setGrantType(tokenReq.getGrantType());
+        accessTokenDO.setIssuedTime(timestamp);
+        accessTokenDO.setTokenBinding(tokReqMsgCtx.getTokenBinding());
+
+        if (OAuth2ServiceComponentHolder.isConsentedTokenColumnEnabled()) {
+            String previousGrantType = validationBean.getGrantType();
+            // Check if the previous grant type is consent refresh token type or not.
+            if (!StringUtils.equals(OAuthConstants.GrantTypes.REFRESH_TOKEN, previousGrantType)) {
+                // If the previous grant type is not a refresh token, then check if it's a consent token or not.
+                if (OIDCClaimUtil.isConsentBasedClaimFilteringApplicable(previousGrantType)) {
+                    accessTokenDO.setIsConsentedToken(true);
+                }
+            } else {
+                /* When previousGrantType == refresh_token, we need to check whether the original grant type
+                 is consented or not. */
+                AccessTokenDO accessTokenDOFromTokenIdentifier = OAuth2Util.getAccessTokenDOFromTokenIdentifier(
+                        validationBean.getAccessToken(), false);
+                accessTokenDO.setIsConsentedToken(accessTokenDOFromTokenIdentifier.isConsentedToken());
+            }
+
+            if (accessTokenDO.isConsentedToken()) {
+                tokReqMsgCtx.setConsentedToken(true);
+            }
+        }
+        return accessTokenDO;
+    }
+
+    private boolean validatePersistedAccessToken(RefreshTokenValidationDataDO validationBean, String clientId)
+            throws IdentityOAuth2Exception {
+
+        if (validationBean.getAccessToken() == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Invalid Refresh Token provided for Client with " + "Client Id : " + clientId);
+            }
+            throw new IdentityOAuth2Exception("Persisted access token data not found");
+        }
+        return true;
+    }
+
+    @Override
+    public boolean isLatestRefreshToken(OAuth2AccessTokenReqDTO tokenReq, RefreshTokenValidationDataDO validationBean,
+                                        String userStoreDomain) throws IdentityOAuth2Exception {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Evaluating refresh token. Token value: " + tokenReq.getRefreshToken() + ", Token state: " +
```

Comment:
Shall we mask/remove these sensitive info such as token values.

Reply:
Here we have tpoo use the util `IdentityUtil.isTokenLoggable(IdentityConstants.IdentityTokens.REFRESH_TOKEN)` and then only log the tokens

Reply:
We can check if the token whether the token is loggable and then log https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java#L148-L157

there is one for refresh token as well
https://github.com/wso2/carbon-identity-framework/blob/master/components/identity-core/org.wso2.carbon.identity.base/src/main/java/org/wso2/carbon/identity/base/IdentityConstants.java#L340-L355

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2091/commits/ff7a124dc1d0d41040fde1134ad8b524f98a8329